base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+required_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
Kindly create the getter and setter method for the `nameable` instance variable. You can use `attr_accessor` to achieve the same.